lua/kulala/parser/headers.lua

```diff
@@ -0,0 +1,85 @@
+--- This class is used to manage the HTTP headers
+--- HTTP header has a case-insensitive name - so class store it in a lower case
+--- and can have a multiple values - which are stored in a table
+---@class HTTPHeaders
+---@field _headers table<string, string[]> - internal table storing the HTTP headers
+local HTTPHeaders = {}
+
+--- new creates a new instance of HTTPHeaders
+---@return HTTPHeaders
+function HTTPHeaders.new()
+  return setmetatable({ _headers = {} }, HTTPHeaders)
+end
+
+--- has checks if a given name exists
+---@param self HTTPHeaders
+---@param key string
+---@return boolean
+function HTTPHeaders.has(self, key)
+  return self._headers[key:lower()] ~= nil
+end
+
+--- has checks if a given name exists and contains a specific value
+---@param self HTTPHeaders
+---@param key string
+---@param value string
+---@return boolean
+function HTTPHeaders.has_value(self, key, value)
+  if not self:has(key) then
+    return false
+  end
+  for _, v in ipairs(self._headers[key:lower()]) do
+    if v == value then
+      return true
+    end
+  end
+  return false
+end
+
+---@param self HTTPHeaders
+---@param key string
+---@return any
+function HTTPHeaders.__index(self, key)
+  return rawget(getmetatable(self), key) or rawget(self._headers, key:lower())
+end
+
+---@param self HTTPHeaders
+---@param key string
+---@param value string
+function HTTPHeaders.__newindex(self, key, value)
```

Comment:
This is an equivalent of https://developer.mozilla.org/en-US/docs/Web/API/Headers/append - dunno if kulala needs more functionality than appending. More methods can be added.